.code-generation/config.js

```diff
@@ -0,0 +1,98 @@
+function CanonicalColorQualifier(payload) {
+  const {qualifierDTO, langConfig} = payload;
+  // This case supports two types of qualifiers
+  // TODO this structure needs to be aligned
+
+
+  if (qualifierDTO.qualifiers && qualifierDTO.name === 'rgb') {
+    return [
+      payload.formatClassOrEnum(qualifierDTO.group, langConfig),
+      langConfig.groupDelimiter,
+      payload.formatMethod(qualifierDTO.name, langConfig),
+      '(',
+      `"${qualifierDTO.qualifiers[0].value}"`.replace('#', ''),
+      ')'
+    ].join('');
+  } else {
+    const simpleColorName = qualifierDTO.qualifiers ?  qualifierDTO.qualifiers[0].name : qualifierDTO.name;
+    return [
+      payload.formatClassOrEnum(qualifierDTO.group, langConfig),
+      langConfig.groupDelimiter,
+      simpleColorName.toUpperCase()
+    ].join('');
+  }
+}
+
+
+module.exports = {
+  "SDKSpecVersion": "master",
+  "langConfig": {
+    lang: 'Kotlin',
+    methodDelimiter: ' ',
+    groupDelimiter: '.',
+    hideActionGroups: false,
+    specialCharacterMapping: [
+      ['$', '\$']
+    ],
+    openQualifiersChar : ' {',
+    closeQualifiersChar : ' }',
+    openActionChar: '(',
+    closeActionChar: ')',
+    unsupportedTxParams: ['fl_waveform', 'fl_animated', 'e_tint', 'u_', 'e_theme', 'l_fetch', 'l_text', 'u_text', 'af_', 'l_video:', 'if_', 'e_fade', 'c_fill', 'palette_f00', 'g_ocr_text'],
```

Comment:
That's a lot! 
no underlay?

Reply:
Same problem with overlay,
We don't have the correct source.format() syntax, this is partially due to the code generator.
I think once we stabilise the code generator and finish aligning the spec, we should revisit this list.